### ISSUE TITLE
Win Probability Modeling in Overtime Issue?

### ISSUE THREAD

**Issue by AccountableRefs** [nflverse/nflfastR#307]
Hello,

Thanks for all the great work on this project! 

I noticed some oddities in the overtime Win Probability modeling. I noticed ties aren't accounted for (and have since read that is a known limitation of the current implementation). But I'm seeing some erratic swings.

Take for instance in OT at 1:47 in 2021_10_DET_PIT. A 4th and 23 punt results in a 0.66+ WP shift. Is this a consequence of not accounting for ties? It seems like the team with possession has a WP that reflects winning the game outright and the defensive team has 1 - wp (effectively giving them credit for their own chance of winning the game and the odds of a game ending in a tie). Is that responsible for these swings?

**Comment by guga31bb** [comment]
Yeah, the OT WP model is garbage. I tried to fix this at one point but it's not easy and gave up before I finished.

And yes the massive WP shifts are because it gives p(win game) for the team with the ball, without accounting for ties- what you said here is exactly right.

**Comment by AccountableRefs** [comment]
Thank you very much for the reply, and sorry for my slow response. Glad to hear I was on the right track. As a bit of an exercise, I wanted to familiarize myself with the nflfastr wp model and tinker with an approach to more accurately predicting OT win probabilities (don't think it will necessarily be fruitful but wanted to give it a shot).

I'm an oddball working in Python and Windows. xgboost is supposed to work almost identically on R/Python and Windows/Linux (minus details like random seed differences). But I'm not able to replicate your modeling predictions by loading the model directly (https://github.com/nflverse/nflfastR-data/blob/master/models/wp.model) or by attempting to recreate the model using the training approach in https://github.com/nflverse/nflfastR/blob/master/data-raw/MODELS.R . I also tried loading https://github.com/nflverse/fastrmodels/blob/621979ae82e74f3bbdedacfaca513882e3e7837a/data/wp_model.rda, but as expected it won't load in Python. 

I was using xgboost to generate predictions for 2020 (limited for qtr <= 4 and down 1-4 to skip the special case processing). I was careful to ensure column orders were identical, generation of the additional wp fields was done correctly, that I omitted spread_time for standard wp predictions, and that I applied all the same filtering. Neither wp.model (very off) and my own model based on models.r (much closer) worked. As far as generating my own model, I'm using the same calibration data listed in Models.R and what I believe are the same params. I believe wp.model and wp_spread.model may be inaccurate because they have different params (as returned by save_config()) than those listed in Models.R and the ones I found by inspecting the contents of wp_model.rda (which exactly match my own when following the Models.R example).

Is there a different model I should be loading or (ideally) a different set of training data and params I should be using to train my own model to replicate your results?

Thanks again!

**Comment by guga31bb** [comment]
I think that model in nflfastR-data is old (sorry, I know it's confusing to find everything). Can you replicate if you use the models stored in here? https://github.com/nflverse/fastrmodels/tree/master/data

**Comment by AccountableRefs** [comment]
Thanks for the quick response!

Unfortunately, that's the rda I tried earlier that won't load in Python. I believe it's essentially a raw memory dump which, as far as I can tell, isn't compatible across versions. I tried the pyreadr and rdata modules to see if they could handle it (they couldn't). I also tried decompressing it and loading it using xgboost as well which did not work. 

But decompressing it allowed me to look at the raw data and find the save_config() settings that were used in the wp_model.rda. Those config settings appear to match the ones from my own attempt to implement the training (I even rolled back from the newest version of xgboost to the 1.1.0 version that I saw in the wp_model.rda config - no luck), so perhaps my issue is the calibration dataset I'm using (I've tried https://github.com/guga31bb/metrics/blob/master/wp_tuning/cal_data.rds and https://github.com/nflverse/nflfastR-data/blob/master/models/cal_data.rds).

Is there an up-to-date version of the wp/wp_spread model not in rda form but exported via save_model?
